R/roles.R

```diff
@@ -307,6 +307,9 @@ update_role <- function(recipe, ..., new_role = "predictor", old_role = NULL) {
 #' @rdname roles
 #' @export
 remove_role <- function(recipe, ..., old_role) {
```

Comment:
```suggestion
remove_role <- function(recipe, ..., old_role, call = caller_env()) {
```

I'm not exactly sure where all this is called, but one level higher up in the call stack seems more likely to be familiar to a user.

Reply:
This function is already the highest level it needs to be as it is user-facing. 

``` r
library(recipes)
data("biomass", package = "modeldata")

recipe(x = biomass) |>
  update_role(sample, new_role = "role1") |>
  remove_role(sample)
#> Error in `remove_role()`:
#> ! argument "old_role" is missing, with no default.
#> Backtrace:
#>     ▆
#>  1. └─recipes::remove_role(...)
#>  2.   └─rlang::abort("argument \"old_role\" is missing, with no default.") at recipes/R/roles.R:311:4
```

<sup>Created on 2023-03-22 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>